interceptor_ext_test.go

```diff
@@ -268,3 +348,46 @@ func (cc *headerInspectingClientConn) Receive(msg any) error {
 	}
 	return err
 }
+
+type httpMethodChecker struct {
+	client bool
+	count  atomic.Int32
+}
+
+func (h *httpMethodChecker) WrapUnary(unaryFunc connect.UnaryFunc) connect.UnaryFunc {
+	return func(ctx context.Context, req connect.AnyRequest) (connect.AnyResponse, error) {
+		h.count.Add(1)
+		if h.client {
+			// should be blank until after we make request
+			if req.HTTPMethod() != "" {
+				return nil, fmt.Errorf("expected blank HTTP method but instead got %q", req.HTTPMethod())
+			}
+		} else {
+			// server interceptors see method from the start
+			if req.HTTPMethod() != http.MethodPost {
```

Comment:
Should we loosen this up? For servers, we just need to know that the method isn't the empty string - depending on protocol, it's fine for it to be POST or GET.

Reply:
OK, makes sense to me!